### ISSUE TITLE
GitHub Code Mappings don't work for my iOS/tvOS Project

### ISSUE THREAD

**Issue by eric** [getsentry/sentry#43516]
### Environment

SaaS (https://sentry.io/)

### Problem

I can't get source code mapping to work on my iOS/tvOS project. This is re-opening #41194 now that I have source uploads working again and have confirmation that the feature should work for iOS/tvOS.

I've setup the Source Code Linking like this:

<img width="653" alt="screenshot-2023-01-20-13 02 36@2x" src="https://user-images.githubusercontent.com/470/213805505-51112790-8565-49ca-941d-311d399d8d6a.png">

And when I view a stack trace that contains this file:

<img width="530" alt="screenshot-2023-01-20-13 02 23@2x" src="https://user-images.githubusercontent.com/470/213805443-d8afade1-9dff-4a92-b839-4e0f3994b252.png">

It does not show the link to GitHub.

The file is available at `https://github.com/fancybits/channels/blob/master/livetv-app/Classes/Controllers/Video%20Player/LTVideoViewController.m` which would be the correct place to map the file with the prefix `/Users/fancybox/source/channels` removed from the path.

**Comment by eric** [comment]
CC @armenzg @philipphofmann

**Comment by Dhrumil-Sentry** [comment]
Hi @eric , Our team is working on this currently and we plan to support stack trace linking for iOS(Swift, Objective C). We expect to have an update in the next 1-2 months

**Comment by eric** [comment]
Did something happen between https://github.com/getsentry/sentry/issues/41194#issuecomment-1320759877 and now? It had worked once at that time and I was told it was working.

What is it that is missing for iOS right now? All of the information seems to be available.

For instance, I can generate the correct URL with the inputs in the screenshot and a bash script:

```
#!/bin/bash

repo="fancybits/channels"
branch="master"
stack_trace_root="/Users/fancybox/source/channels/"

file_path="$(echo $1 | sed -e "s/${stack_trace_root//\//\\/}//g")"

echo "https://github.com/$repo/blob/$branch/$file_path"
```

```
$ url-example.sh '/Users/fancybox/source/channels/livetv-app/Classes/Controllers/Video Player/LTVideoViewController.m'
https://github.com/fancybits/channels/blob/master/livetv-app/Classes/Controllers/Video Player/LTVideoViewController.m
```

**Comment by armenzg** [comment]
Hi @eric could you please point me to your org slug or directly to an issue?

Do you see stacktrace-link being hit?
I would expect it to be something like:
`https://sentry.io/api/0/projects/<your_org>/<your_project_slug>/stacktrace-link/?absPath=/Users/fancybox/source/channels/livetv-app/Classes/Controllers/Video Player/LTVideoViewController.m&file=Video Player/LTVideoViewController.m` (maybe few more values but I'm mostly interested in `absPath` and `file`).

I would expect this to work with blank stack root and blank source code root.

**Comment by eric** [comment]
@armenzg Here's a trace I would expect to link to GitHub but doesn't: https://fancy-bits-llc.sentry.io/issues/2774290541/

**Comment by armenzg** [comment]
Hi @eric 
The call to stacktrace-link has `file=LTGuideAiringCell.m` rather than the full path, thus, not matching the stack trace root. Here's [the code evaluation](https://github.com/getsentry/sentry/blob/446c923c26213b48cb0373e43ab2aa3ac3b82d3c/src/sentry/api/endpoints/project_stacktrace_link.py#L219) I'm referring to.

I wonder if the SDK has a bug that should give us the correct path or if we should consider the absPath besides the file argument.

[I've filed it](https://getsentry.atlassian.net/browse/WOR-2836) and will discuss with the team.

Thanks @eric for your persistence!

**Comment by eric** [comment]
So is this an issue with the symbolicator? Where does that code live?

**Comment by eric** [comment]
> I wonder if the SDK has a bug that should give us the correct path or if we should consider the absPath besides the file argument.

I definitely expected for the `absPath` to be used here, which was the reason for me specifying the settings I did in the screenshot above.

**Comment by armenzg** [comment]
> So is this an issue with the symbolicator? Where does that code live?

When you load any issue with a stack trace, you will see a call to the `stacktrace-link` API.

**Comment by eric** [comment]
Could the symbolicator be the thing that is providing the `path` and `absPath` on iOS and tvOS?

**Comment by eric** [comment]
I still would like links to GitHub to work.

**Comment by philipphofmann** [comment]
@armenzg, you answered a couple of questions on this issue. Do you maybe have an update for @eric?

**Comment by armenzg** [comment]
> @armenzg, you answered a couple of questions on this issue. Do you maybe have an update for @eric?

My apologies. I was unavailable last week. I'm asking for input on this.

**Comment by malwilley** [comment]
I took some time to look into this, so I'll try to record everything I found here:

Here is an issue I’ll use as an example to explain what is wrong with the current state of things: [Internal link to a test issue](https://sentry-sdks.sentry.io/issues/4590231710/). And here is the JSON for a frame we'd like to add code mappings for:

```json
{
	"function": "main",
	"symbol": "main",
	"package": "/private/var/containers/Bundle/Application/01EDD90B-311B-4C40-AB57-36CAC0693F26/iOS-Swift.app/iOS-Swift",
	"filename": "AppDelegate.swift",
	"abs_path": "/Users/runner/work/sentry-cocoa/sentry-cocoa/Samples/iOS-Swift/iOS-Swift/AppDelegate.swift",
	"lineno": 5,
	"in_app": true
}
```
For this frame, we expect the frame package to be something like `iOS-Swift`, but the actual value is `/private/var/containers/Bundle/Application/55283E74-25F5-41CA-82A1-985C4E4B374F/iOS-Swift.app/iOS-Swift`. Here [in the code](https://github.com/getsentry/sentry/blob/master/src/sentry/utils/event_frames.py#L82-L95) we use package to transform the `abs_path` from `/Users/runner/work/sentry-cocoa/sentry-cocoa/Samples/iOS-Swift/iOS-Swift/AppDelegate.swift` to `iOS-Swift/iOS-Swift/AppDelegate.swift` which matches the source code better. However, since package is a large file path itself, the current logic does not work as intended.

I've confirmed with the SDK folks that `package` has the correct value, so our app code will need to change. It appears that we should drop the relative path logic and let the user map the full value in `abs_path`. This will be the simplest method for users to understand and as long as the root portion of `abs_path` does not change, will be stable. Another consideration is that some customers may have found a way to get this mapping working under the current logic, so we will need to find out if that is the case and migrate those users to the new simplified mapping logic.
